.github/actions/changes/action.yml

```diff
@@ -0,0 +1,52 @@
+name: "Github Changes"
+description: "Compute by any means necessary the changes attempting to be merged, or that have been merged in to a branch."
+inputs:
+  workflow-file:
+    description: "local repository path to the workflow file, no leading slash."
+    required: true
+outputs:
+  changes-base-githash:
+    description: The shared git hash of a pr, a bors target branch, or the last successful build of the workflow-file on the branch.
+    value: ${{ steps.calc.outputs.changes-base-githash }}
+  changes-target-branch:
+    description: If a target branch exists for either bors or a pr, the branch.
+    value: ${{ steps.calc.outputs.changes-target-branch }}
+  changes-pull-request-number:
+    description: If a pullrequest number can be found, either in bors, or in a pr, the number.
+    value: ${{ steps.calc.outputs.changes-pull-request-number }}
+runs:
+  using: "composite"
+  steps:
+    - id: calc
+      run: |
+        if [[ $(git log --pretty=format:"%H" -n 26 | wc -l) < 2 ]]; then
```

Comment:
This could use a little bit of love, like all the actions I put together over break.
1. how do we want to log?
2. flag for logging?
3. The shell inside these actions scripts needs refactoring.   I just got it working but not happy with the state.  Targeting getting it working before getting it pretty.

That said, put real effort in to the API exposed by the actions to the workflows.   Look at that as primary take away from this for now.